ts/src/test/test.ts

```diff
@@ -52,534 +34,568 @@ const exchange = new (ccxt)[exchangeId] ({
-for (const error of filteredErrors) {
-    // eslint-disable-next-line global-require, import/no-dynamic-require, no-path-concat
-    const errorTest = await import (__dirname + '/errors/test.' + error + '.js');
-    testFiles[error] = errorTest['default'];
-}
-
-//-----------------------------------------------------------------------------
-
-const keysGlobal = 'keys.json';
-const keysLocal = 'keys.local.json';
-const keysFile = fs.existsSync (keysLocal) ? keysLocal : keysGlobal;
-const settingsFile = fs.readFileSync (keysFile);
-// eslint-disable-next-line import/no-dynamic-require, no-path-concat
-let settings = JSON.parse (settingsFile);
-settings = settings[exchangeId];
-if (settings) {
-    const keys = Object.keys (settings);
-    for (let i = 0; i < keys.length; i++) {
-        const key = keys[i];
-        if (settings[key]) {
-            settings[key] = ccxt.deepExtend (exchange[key] || {}, settings[key]);
-        }
+properties.push ('loadMarkets');
+for (let i = 0; i < properties.length; i++) {
+    const property = properties[i];
+    const filePath = __dirname + '/Exchange/test.' + property + '.js';
+    if (fs.existsSync (filePath)) {
+        // eslint-disable-next-line global-require, import/no-dynamic-require, no-path-concat
+        testFiles[property] = await import (pathToFileURL(filePath)) as any;
     }
 }
+import errorsHierarchy from '../base/errorHierarchy.js';
+Object.keys (errorsHierarchy)
+    // eslint-disable-next-line no-path-concat
+    .filter ((error) => fs.existsSync (__dirname + '/errors/test.' + error + '.js'))
+    .forEach (async (error) => {
+        // eslint-disable-next-line global-require, import/no-dynamic-require, no-path-concat
+        testFiles[error] =  await import (__dirname + '/errors/test.' + error + '.js');
+    });
 
-Object.assign (exchange, settings);
+const AuthenticationError = errorsHierarchy.AuthenticationError;
 
-// check auth keys in env var
-const requiredCredentials = exchange.requiredCredentials;
-for (const [credential, isRequired] of Object.entries (requiredCredentials)) {
-    if (isRequired && exchange[credential] === undefined) {
-        const credentialEnvName = (exchangeId + '_' + credential).toUpperCase (); // example: KRAKEN_APIKEY
-        const credentialValue = process.env[credentialEnvName];
-        if (credentialValue) {
-            exchange[credential] = credentialValue;
-        }
-    }
-}
+// non-transpiled commons
+const rootDir = __dirname + '/../../../';
+const envVars = process.env;
 
-if (settings && settings.skip) {
-    console.log ('[Skipped]', { 'exchange': exchangeId, 'symbol': exchangeSymbol || 'all' });
-    process.exit (0);
+class emptyClass {}
+
+function dump (...args) {
+    console.log (...args);
 }
-if (exchange.alias) {
-    console.log ('[Skipped alias]', { 'exchange': exchangeId, 'symbol': exchangeSymbol || 'all' });
-    process.exit (0);
+function io_file_exists (path) {
```

Comment:
why snake case in typescript files? 🤔

Reply:
we didn't have our unCamelCase https://github.com/ccxt/ccxt/pull/13882 transpile update, so our transpiler didnt handle uncamecaling automatically for functions, thus, while transpiling `test.js -> test.py/php`, these methods where not uncamelcased automatically, so i've avoided burden of transpiler adjustements for it. but if we should, lmk and i'll push.

Reply:
@ttodua  it is easy to make the transpiler uncamelcase the names

Reply:
ok, give me a while.
I've almost finished other things too here.

Reply:
let me know if you need help with the unCamelCase option